### ISSUE TITLE
Windows and Mac OSX installer

### ISSUE THREAD

**Issue by pemensik** [raceintospace/raceintospace#249]
I have pushed simple change in commit c109b0872d0bebe1611e691a30fa0f9bde6fabc2 to include CPack in cmake project. It adds targets package and package_source to build directory. First is binary package. There is support for Windows installer NSIS, which might be handy. Similar approach could be used to produce package for Mac OSX. It can do more formats, but it seems unusable to me for rpm.

Would someone with Windows development tools try to produce installer, whether it seems handy? Do we want to produce easy to install binaries for public audience?

Source package target is not very useful, reworked make dist is far superior. It packages almost everything it cames along, so it has to be done only to empty source tree.

**Comment by rnyoakum** [comment]
Part of this problem is getting the game to work on OSX. There's a closed issue #150 which described the problem, and got closed before I learned the problem wasn't fully fixed; apparently the game will build but not run properly on OSX. I don't have access to a Mac, so I haven't been able to debug it.

**Comment by rodneyrod** [comment]
What would you need done to help? I have a mac I can run tests with.

**Comment by peyre** [comment]
Speaking of which, when we're ready to put out a release, would someone with a Mac let me know where the config file is stored?  I know where it is on Windows and Linux, but not on the Mac.

**Comment by rodneyrod** [comment]
SDL_config?

**Comment by rnyoakum** [comment]
Thanks for the offer! The problem, as it was described in issue #150, is

> When I compile and run on my machine (macOS mojave) the app screen is blank - the music plays, but you cannot interact with the game and the screen stays blank. I did install SDL from brew since cmake didnt seem to want to download and build that.

Both @bmoscon and @tusharpm apparently experienced this issue, but I couldn't help without a machine to test on. There's some debug information they provided.

**Comment by peyre** [comment]
No, I'm sorry.  The file named "config" that stores Advanced Preferences.  It's in ~/.raceintospace, on Linux systems.

**Comment by pemensik** [comment]
> Speaking of which, when we're ready to put out a release, would someone with a Mac let me know where the config file is stored? I know where it is on Windows and Linux, but not on the Mac.

I know very little about MacOS, not really its owner. According to [stackoverflow](https://stackoverflow.com/questions/410013/where-do-osx-applications-typically-store-user-configuration-data) good path would be */Users/username/Library/Preferences/raceintospace* But we should ask someone with Mac OS development experience.

**Comment by peyre** [comment]
That makes sense, given how closely related the Mac back end is to Linux.  I wonder, is ~/ used as a shorthand for /Users/username, like it is for /home/username in Linux?

**Comment by pemensik** [comment]
~/ is expanded by bash on GNU/Linux. I think standard library or kernel itself does not know anything about it. We should use getenv("HOME"), but blind guessing it just wrong. We have here some people with macs. I will ask, maybe someone tried to develop on it. Hopefully someone would help.

**Comment by rodneyrod** [comment]
Config file is at ~/.raceintospace
Normally most MacOS games would put these in ~/Library/Application Support/raceintospace but that's not a blocker.

**Comment by rodneyrod** [comment]
I should also add the old MacOS raceintospace binary currently on the download site isn't compatible with 10.15+ and will have to be recompiled as 64bit.

**Comment by peyre** [comment]
Thanks rodney!  I'll add that bit to my guide to the game.

It sounds like you might be a programmer.  If you're interested in working on the game, shoot me a note at lordpeyre at yahoo.com.

**Comment by pemensik** [comment]
well modern Linux distributions should use ${XDG_CONFIG:-$HOME/.config}/raceintospace for configuration. But that would be nice to cleanup if we know all platforms work.

By the way, I have made *make run* target, which does run the game even without installing. I think it would be useful for simpler start of the game.

**Comment by ryandesign** [comment]
The blank window problem on Mojave and later is just [a libsdl bug](https://bugzilla.libsdl.org/show_bug.cgi?id=4788) that has [now been fixed](https://hg.libsdl.org/SDL/rev/ab7529cb9558), but there has not yet been a new release of libsdl containing that fix.

You can [use MacPorts to install raceintospace](https://ports.macports.org/port/raceintospace). The libsdl bug [should be fixed there soon](https://github.com/macports/macports-ports/pull/7314) too.

**Comment by pemensik** [comment]
I guess we should finally prepare usable installer and make a release. Do we have anyone capable testing recent version on a Mac, whether it is useable?

**Comment by sflynn-rightpoint** [comment]
I've got an M1 Macbook Pro I could use, running Monterey.

**Comment by sflynn850** [comment]
I've got an M1 Macbook Pro running Monterey that I could use?

**Comment by peyre** [comment]
Hey this is exciting!  A new release after all this time.  Pace has said he could do, or help with, the Linux release.
